packages/client/src/core/engine.ts

```diff
@@ -150,34 +191,52 @@ export class Engine {
       } = calcPosition(enemy)
 
       if (
-        playerLeft < enemyRight &&
-        playerRight > enemyLeft &&
-        playerTop < enemyBottom &&
-        playerBottom > enemyTop
+        playerLeft + this.collisionOffset < enemyRight &&
+        playerRight - this.collisionOffset > enemyLeft &&
+        playerTop + this.collisionOffset < enemyBottom &&
+        playerBottom - this.collisionOffset > enemyTop
       ) {
         this.gameOver = true
       }
     })
   }
 
   private handleEnemy = (ctx: CanvasRenderingContext2D, deltaTime: number) => {
-    if (this.enemyTimer > this.enemyInterval + this.randomEnemyInterval) {
-      const random = randomFromInterval(0, this.enemiesParams.length - 1)
-      const enemyParams = this.enemiesParams[random]
-
-      this.enemies.push(
-        new Enemy({
-          gameWidth: this.gameWidth,
-          gameHeight: this.gameHeight,
-          width: enemyParams.width,
-          height: enemyParams.height,
-          speed: 5,
-          imageSrc: enemyParams.imageSrc,
-          y: enemyParams.y ? enemyParams.y() : undefined,
+    if (this.enemyTimer > this.enemyInterval + randomFromInterval(100, 1000)) {
+      const enemyType = randomFromInterval(0, 2) as 0 | 1 | 2
+
+      const enemyByType: Record<0 | 1 | 2, Enemy> = {
```

Comment:
[Исправление]
Лучше создать тип `type EnemyType = 0 | 1 | 2`
Тогда тут будет так:
```suggestion
      const enemyType: EnemyType = randomFromInterval(0, 2)

      const enemyByType: Record<EnemyType, Enemy> = {
```

Reply:
Сделал, но от as все равно не получилось избавится :( 
тк randomFromInterval возвращает  просто number и TS ругается что
 Type 'number' is not assignable to type 'EnemyType'.
```
      const enemyType: EnemyType = randomFromInterval(0, 2) as EnemyType

      const enemyByType: Record<EnemyType, Enemy> = { 
```

Reply:
Оберни `randomFromInterval(0, 2)` в функцию, которая будет возвращать `EnemyType`. Использование as = выключить TS, так лучше не делать

Reply:
<img width="630" alt="image" src="https://github.com/rainstr7/blade-runners.game/assets/124124244/4b696b65-97b6-4b44-b780-4d931a89238a">

Я видимо чтото не понимаю?